### ISSUE TITLE
Issue with indexing after add/remove using symfony

### ISSUE THREAD

**Issue by rokasmi** [ninsuo/symfony-collection#42]
After adding/removing collection form, all the rest forms indexes are being reindexed.
Why?
That causes issue with handling request data in symfony. Request arrays keys doesnt correctly coincidence with entities after handleRequest().

I have written the problem in [stackoverflow](http://stackoverflow.com/questions/39638354/form-collection-data-after-handlerequest-is-swapping/39639856#39639856)
The correct way would be not reindex forms after add/delete etc. See symfony example: [http://jsfiddle.net/847Kf/4/](http://jsfiddle.net/847Kf/4/)

**Comment by ninsuo** [comment]
Reindexing is useul when you care about your element's position. Else, "moving up" and "moving down" would change the elements position in the UI, but refreshing would give the old positions back. 

The demo website contain a Doctrine usage example:
http://symfony-collection.fuz.org/advanced/usageWithDoctrine

Thanks

**Comment by rokasmi** [comment]
But then you cant use this library for entities which has references to other tables. Because you will mess up your records relations.

**Comment by ninsuo** [comment]
Yes, you are right. 
I may add a `no-reindex` option to disable reindexing when adding/deleting elements.
This will obviously disable lots of options but this is doable.

**Comment by rokasmi** [comment]
Great! At this moment, I think, it may be useful to write a notice in [doctrine example](http://symfony-collection.fuz.org/advanced/usageWithDoctrine), that you should not use this library if yours collections entities have relations or it is important to keep ids same.

**Comment by mktcode** [comment]
Here are a few thoughts on this:

Since ordering will mostly (if not always) be done server side by adding an 'order' property to the entity, I would suggest to simply provide an option to update such a field.

```
$('#my-collection').collection({
    ...
    order_field_selector: '.order'
});
```

```
$builder
    ...
    ->add('order', null, ['attr' => ['class' => 'order']])
```
Now, after manipulating the collection you only have to refill these fields based on the current order in the ui:
```
if (settings.order_field_selector) {
    elements = collection.find(settings.elements_selector);
    elements.each(function(i, item) {
        $(item).find(settings.order_field_selector).val(elements.index($(item)));
    });
}
```
Swapping the contents of the fields rather than the swapping the entire entry forms should be rethought in general. It just seems way to expensive and prone to me. Also for me this plugin makes really not much sense, as long as it does not support collection entities having relations or simply a consistent id.

**Comment by ninsuo** [comment]
Thank you for your feedback. I'm quite busy in fact so I don't spend that much time on this project.

You can push some issues with more efficiency by implementing some samples that doesn't work yet in the demo website. There I can have a look very quickly and find a way to handle the case.

**Comment by Quovandius** [comment]
Hello,

Sorry for disturbing, but i don't understand if you have found a solution for this issue, its a real BIG problem for me. 

For exemple : i have create 3 sub entity with collection form, i submit the form, its ok.
Then, i edit the entity and i delete the 1st sub form of collection. When i submit the update, all index are refresh .....

Any solution is existing for block this ?

Thanks for your help.

**Comment by mktcode** [comment]
Yes, I solved this issue in my fork: https://github.com/mktcode/symfony-collection

See the "Ordering" part in the README. You can use my fork by adding

`"symfony-collection": "https://github.com/mktcode/symfony-collection.git#^2.0.5"`

to your bower dependencies, since I have not registered my fork as a separate bower package because I would prefer if it gets merged into this main repo.

...or you can of course just download the archive: https://github.com/mktcode/symfony-collection/archive/2.0.5.zip

**Comment by Quovandius** [comment]
Impressive =) nice work
Thanks for your quick answer.

I have another question, i ask it here ^^
Is it possible to disable this feature ? i don't care to reorder my subform for any reason ^^

**Comment by mktcode** [comment]
This is functionality was already included in this main repository. Just add the `allow_up: false` and `allow_down: false` options.

**Comment by Quovandius** [comment]
Yes i have found these options. Theses are very useful.

My explanations are not clear. I try to clear what i mean : 

-  I create an entity with form collection, and i add 3 sub entity -- Ok
- Then, i edit this entity, and delete one sub entity, i submit -- Ok
- Then, i edit again my entity, i delete the second sub entity, and add a new line (this line is empty), i submit -- OK .... problem ^^

For this last action, the submit form is ok .... BUT on database, the second sub entity is cleared ... empty ...

I search a tips for generate an error or disable the delete button with a filter / callback / event. In some case, i need to block this, and just want to add a new sub form. If i block the delete option, its for all sub entities.

This behaviour is dangerous in my case (i have created an order form, with order line so ^^).

Thanks for your help, i have try your fork, its a very good work =)

**Comment by ninsuo** [comment]
There's already a pending fix for this issue ( #48 ).
I've no time to review it right now, but you can implement it from @mktcode's [repository](https://github.com/mktcode/symfony-collection).

Don't hesitate to give your feedback here once you succeeded.

**Comment by ninsuo** [comment]
@mktcode i'll try my best to find an hour or two to review your PR asap

**Comment by ninsuo** [comment]
Fixed in #59
